doc/lua_api.txt

```diff
@@ -8603,8 +8602,221 @@ Used by `minetest.add_particlespawner`.
         -- If set to a valid number 1-6, specifies the tile from which the
         -- particle texture is picked.
         -- Otherwise, the default behavior is used. (currently: any random tile)
+
+        -- Legacy definition fields
+
+        minpos = {x=0, y=0, z=0},
+        maxpos = {x=0, y=0, z=0},
+        minvel = {x=0, y=0, z=0},
+        maxvel = {x=0, y=0, z=0},
+        minacc = {x=0, y=0, z=0},
+        maxacc = {x=0, y=0, z=0},
+        minexptime = 1,
+        maxexptime = 1,
+        minsize = 1,
+        maxsize = 1,
+        -- The particles' properties are random values between the min and max
+        -- values.
+        -- applies to: pos, velocity, acceleration, expirationtime, size
+        -- If `node` is set, min and maxsize can be set to 0 to spawn
+        -- randomly-sized particles (just like actual node dig particles).
+    }
+
+### Modern definition fields
+After v5.5.0, spawner properties can be defined in several different ways depending
+on the level of control you need. `pos` for instance can be set as a single vector,
+in which case all particles will appear at that exact point throughout the lifetime
+of the spawner. Alternately, it can be specified as a min-max pair, specifying a
+cubic range the particles can appear randomly within. The following definitions are
+all equivalent, listed in order of precedence from lowest to highest (if both a
+lower-precedence and higher-precedence field are present, the lower-precedence field
+will be ignored):
+
+    {
+	    maxpos = {x = 0, y = 0, z = 0};
+	    minpos = {x = 0, y = 0, z = 0};
+        -- old syntax
+
+        pos = 0;
+        -- all components of every particle's position vector will be set to this
```

Comment:
> tbh I don't think forcing modders to write out vectors is so bad that this should be supported, other devs may disagree

I agree. If anything, there should be a new vector utility constructor, but that's for a different PR.

Reply:
there's a specific reason i allowed for this syntax; the `pos` one is just for sake of example and not very useful. however, there are some properties, like radius and drag, that in the majority of situations you're only ever going to want a unifom value. however, there are rarer cases when e.g. defining drag as a vector is useful, say to avoid interfering with gravity applied by acc. i didn't think it was reasonable to force the extra syntactic salt on the majority of modders just to give an extra feature to a minority, but if the consensus is that this is uncool i'll rip it out?

Reply:
so, is the consensus that i should remove this?

Reply:
Hard to say without at least one other coredev having looked at it.
Removing it in the end after the rest is all ready wouldn't be a problem, would it?

Reply:
don't think so, just means removing a few lines of code in the lua api and changing the docs